app/event/[id]/event-feed.tsx

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useRef, useState } from "react";
+import { useSuspenseQuery } from "@tanstack/react-query";
+import { useAuth } from "@clerk/nextjs";
+import Link from "next/link";
+import { PollCard } from "@/components/cards/social-feed/poll-card";
+import { fakePolls } from "@/app/event/[id]/fake-polls";
+import { PostCard } from "@/components/cards/social-feed/post-card";
+import { AvatarWithLoaderAndFallback } from "@/components/common/Avatar";
+import { userAddressOptions, userOptions } from "@/lib/queries/user";
+import { Textarea } from "@/components/shadcn/textarea";
+import { cn } from "@/lib/tailwind";
+
+function FeedInput({ className }: { className?: string }) {
+  const { getToken, userId } = useAuth();
+  const { data: address } = useSuspenseQuery(
+    userAddressOptions(getToken, userId),
+  );
+  const { data: user } = useSuspenseQuery(userOptions(address));
+
+  // Auto shrink and grow textarea
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const [textareaValue, setTextAreaValue] = useState("");
+  const textareaMinHeight = 54;
+  const textareaMaxHeight = 300;
+  useEffect(() => {
+    const textarea = textareaRef.current;
+    if (!textarea) return;
+    textarea.style.height = `${textareaMinHeight}px`;
+    textarea.style.height = `${textarea.scrollHeight}px`;
+  }, [textareaValue]);
+
+  return (
+    <div className={cn("flex flex-row items-center gap-4", className)}>
+      {user?.avatarUri && (
+        <Link href="/settings">
+          <AvatarWithLoaderAndFallback user={user} />
+        </Link>
+      )}
+      <Textarea
+        ref={textareaRef}
+        onChange={(evt) => setTextAreaValue(evt.target.value)}
+        className={`cursor-pointer border-0 min-h-[${textareaMinHeight}px] focus-visible:ring-transparent rounded-xl text-base text-sm px-4 py-3 w-full max-h-[${textareaMaxHeight}px] placeholder:text-primary-color  text-lg hover:bg-neutral-700`}
+        placeholder={"Don't be shy, say something!"}
+      />
+    </div>
+  );
+}
+
+export function EventFeed({
+  // id,
+  // userId,
+  isDescExpanded,
+}: {
+  // id: string;
+  // userId: string | null;
+  isDescExpanded: boolean;
+}) {
+  // Getting --background value. Used for sticky FeedInput background
+  const [bgColor, setBgColor] = useState("");
+  useEffect(() => {
+    const root = document.documentElement;
+    const computedStyle = getComputedStyle(root);
+    const cssVar = computedStyle.getPropertyValue("--background").trim();
+    setBgColor(`hsl(${cssVar} / .9)`);
+  }, []);
+
+  //  Stuff used to stick FeedInput when scroll bellow of it
+  const containerRef = useRef<HTMLDivElement>(null);
+  const [containerWidth, setContainerWidth] = useState(0);
```

Comment:
we shouldn't need this kind of things with the full power of css, this looks like react-native thinking